c10/core/TensorOptions.h

```diff
@@ -313,17 +301,6 @@ struct C10_API TensorOptions {
                               : c10::nullopt;
   }
 
-  /// Returns the `is_variable` property of the `TensorOptions`.
-  bool is_variable() const noexcept {
```

Comment:
I'm not convinced we should release like this (it's good that you did this to make sure internal things don't call this).  I'm actually not 100% sure what the equivalent should be (how does variable_factories work?), but it might be something like change the default and assert if a user ever tries to set it to False.

Reply:
> I'm actually not 100% sure what the equivalent should be (how does variable_factories work?)

Post this PR, it's always true. You always get a variable no matter what you do.

I agree this is modestly BC-breaking (and I had to fix some sites in fbcode where `is_variable` was being set), but I think this method never really made any sense for end users (it was more of a situation where people sprinkled it in just get things to work, without understanding what it did) and I think it's clearer if we just force people to delete it from their code. At least, there were very *few* fbcode fixes I had to do, and I use that as a proxy for external usage.